File: deamhome/src/main/java/sosteam/deamhome/domain/order/entity/Order.java
package sosteam.deamhome.domain.order.entity;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import sosteam.deamhome.domain.account.entity.Account;
import sosteam.deamhome.domain.item.entity.Item;

import java.util.List;

@Document
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Order {

    @NotNull
    private String address;

    @NotNull
    private String userName;

    @NotNull
    private String email;

    @NotNull
    private String phone;

    private String adminMemo;
    private String content;
    private boolean isPaid = false;
    private boolean isRefund = false;

    @DBRef(lazy = true)
    private Account account;

    @DBRef(lazy = true)
    private List<Item> items;

    public List<Item> addItem(Item item) {
        items.add(item);

        return items;
    }
}

File: src/main/kotlin/sosteam/deamhome/domain/order/entity/Order.kt
package sosteam.deamhome.domain.order.entity

import com.github.f4b6a3.ulid.UlidCreator
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import sosteam.deamhome.global.attribute.OrderStatus
import sosteam.deamhome.global.entity.BaseEntity
import java.time.OffsetDateTime

@Table("order_shop")
class Order(
	@Id
	var id: Long?,
	val address: String,
	val mainDoorPassWord: String,
	val deliveryRequest: String?,
	val userName: String,
	var userNickname: String,
	// 수신인
	var addressee: String,
	val email: String,
	val phone: String,
	var memo: String?,
	val userId: String,
	var totalPrice: Int,
	val content: String?,
	var payType: String,
	val appNo: String,
	val bankId: String,
	val orderDateTime: OffsetDateTime,
	val paymentDateTime: OffsetDateTime?,
	val releaseDateTime: OffsetDateTime?,
	val refundDateTime: OffsetDateTime?,
	val deliveryDateTime: OffsetDateTime?,
	val approveDatetime: OffsetDateTime?,
	val couponIds: List<String?>?,
	val pointPrice: Int,
	var shippingCompany: String?,
	val downloadDays: OffsetDateTime,
	val reasonRefund: String?,
) : BaseEntity() {
	val publicId: String = UlidCreator.getMonotonicUlid().toString().replace("-", "")
	var orderStatus: OrderStatus = OrderStatus.PENDING
	var adminMemo: String? = null
	var isPaid = false
	var isRefund = false
}